src/AlgebraicGeometry/Schemes/Divisors/AlgebraicCycles.jl

```diff
@@ -534,6 +558,25 @@ function integral(W::AbsAlgebraicCycle; check::Bool=true)
   return result
 end
 
+# Getters for the components as honest algebraic cycles, not ideal sheaves.
+function components(::Type{T}, D::AbsAlgebraicCycle) where {T <: AbsAlgebraicCycle}
```

Comment:
It seems to me the signature is lying a bit, since we do not get back an object of type `T`. 
But for usability reasons I can live with it. How do other parts of Oscar handle this?

Reply:
I think, this is the only feasible way. It should be legitimate to specify a non-concrete type as input for the first argument. Maybe we can add some type assertion?